gef.py

```diff
@@ -4308,9 +4308,42 @@ def do_invoke(self, argv):
         new_output = argv[1]
 
         disable_context()
-        res = gdb.execute("""call open("{:s}", 66, 0666)""".format(new_output), to_string=True)
-        # Output example: $1 = 3
-        new_fd = int(res.split()[2], 0)
+
+        if ':' in new_output:
+            address = socket.gethostbyname(new_output.split(':')[0])
+            port = int(new_output.split(':')[1])
+
+            # AF_INET = 2, SOCK_STREAM = 1
+            res = gdb.execute("""call socket(2, 1, 0)""", to_string=True)
+            new_fd = int(res.split()[2], 0)
```

Comment:
Why base 0? Shouldn't the result be in a consistent base? Or is it dependent on the radix setting.

Reply:
This was already here before so I figured that I should just follow.